drivers/can/can_mcp25xxfd.h

```diff
@@ -0,0 +1,605 @@
+#define MCP25XXFD_TEF_FIFO_SIZE       (MCP25XXFD_TEF_FIFO_ITEMS * MCP25XXFD_TEF_FIFO_ITEM_SIZE)
+
+#define MCP25XXFD_TX_QUEUE_START_ADDR MCP25XXFD_TEF_FIFO_SIZE
+#define MCP25XXFD_TX_QUEUE_ITEMS      CONFIG_CAN_MCP25XXFD_MAX_TX_QUEUE
+#define MCP25XXFD_TX_QUEUE_SIZE       (MCP25XXFD_TX_QUEUE_ITEMS * MCP25XXFD_TX_QUEUE_ITEM_SIZE)
+
+#define MCP25XXFD_RX_FIFO_START_ADDR (MCP25XXFD_TX_QUEUE_START_ADDR + MCP25XXFD_TX_QUEUE_SIZE)
+#define MCP25XXFD_RX_FIFO_SIZE_MAX   (MCP25XXFD_RAM_SIZE - MCP25XXFD_RX_FIFO_START_ADDR)
+#define MCP25XXFD_RX_FIFO_ITEMS_MAX  (MCP25XXFD_RX_FIFO_SIZE_MAX / MCP25XXFD_RX_FIFO_ITEM_SIZE)
+
+#define MCP25XXFD_RX_FIFO_ITEMS CONFIG_CAN_MCP25XXFD_RX_FIFO_ITEMS
+#define MCP25XXFD_RX_FIFO_SIZE  (MCP25XXFD_RX_FIFO_ITEMS * MCP25XXFD_RX_FIFO_ITEM_SIZE)
+
+#define MCP25XXFD_RX_FIFO_IDX 1
+#define MCP25XXFD_REG_SIZE    4
+
+#define MCP25XXFD_CRC_POLY 0x8005
+#define MCP25XXFD_CRC_SEED 0xffff
+
+BUILD_ASSERT(MCP25XXFD_TEF_FIFO_SIZE + MCP25XXFD_TX_QUEUE_SIZE +
+	     MCP25XXFD_RX_FIFO_SIZE <= MCP25XXFD_RAM_SIZE,
+	     "Cannot fit FIFOs into RAM");
+
+struct mcp25xxfd_mailbox {
+	can_tx_callback_t cb;
+	void *cb_arg;
+};
+
+#define MCP25XXFD_SPI_CMD_LEN       2
+#define MCP25XXFD_SPI_LEN_FIELD_LEN 1
+#define MCP25XXFD_SPI_CRC_LEN       2
+
+/* MCP25XXFD Opcodes */
+#define MCP25XXFD_SPI_INSTRUCTION_RESET    0x0000
+#define MCP25XXFD_SPI_INSTRUCTION_WRITE    0x2000
+#define MCP25XXFD_SPI_INSTRUCTION_READ     0x3000
+#define MCP25XXFD_SPI_INSTRUCTION_READ_CRC 0xb000
+
+/* MCP25XXFD Operation Modes */
+#define MCP25XXFD_OPMODE_NORMAL_CANFD  0b000
+#define MCP25XXFD_OPMODE_SLEEP         0b001
+#define MCP25XXFD_OPMODE_INT_LOOPBACK  0b010
+#define MCP25XXFD_OPMODE_LISTEN_ONLY   0b011
+#define MCP25XXFD_OPMODE_CONFIGURATION 0b100
+#define MCP25XXFD_OPMODE_EXT_LOOPBACK  0b101
+#define MCP25XXFD_OPMODE_NORMAL_CAN2   0b110
+#define MCP25XXFD_OPMODE_RESTRICTED    0b110
+
+#define MCP25XXFD_WFT_T00FILTER 0b00
+#define MCP25XXFD_WFT_T01FILTER 0b01
+#define MCP25XXFD_WFT_T10FILTER 0b10
+#define MCP25XXFD_WFT_T11FILTER 0b11
+
+#define MCP25XXFD_TDCMOD_AUTO     0b10
+#define MCP25XXFD_TDCMOD_MANUAL   0b01
+#define MCP25XXFD_TDCMOD_DISABLED 0b00
+
+#define MCP25XXFD_TIMING_MIN_INITIALIZER				\
+	{								\
+		.sjw = 1,						\
+		.prop_seg = 0,						\
+		.phase_seg1 = 2,					\
+		.phase_seg2 = 1,					\
+		.prescaler = 1,						\
+	}
+
+#define MCP25XXFD_TIMING_MAX_INITIALIZER				\
+	{								\
+		.sjw = 128,						\
+		.prop_seg = 0,						\
+		.phase_seg1 = 256,					\
+		.phase_seg2 = 128,					\
+		.prescaler = 256,					\
+	}
+
+#define MCP25XXFD_TIMING_DATA_MIN_INITIALIZER				\
+	{								\
+		.sjw = 1,						\
+		.prop_seg = 0,						\
+		.phase_seg1 = 1,					\
+		.phase_seg2 = 1,					\
+		.prescaler = 1,						\
+	}
+
+#define MCP25XXFD_TIMING_DATA_MAX_INITIALIZER				\
+	{								\
+		.sjw = 16,						\
+		.prop_seg = 0,						\
+		.phase_seg1 = 32,					\
+		.phase_seg2 = 16,					\
+		.prescaler = 256,					\
+	}
```

Comment:
These are only used once. Please move them to where they are used instead of using macros. The macros used in other drivers are due to these having multiple front-ends. I don't see that as a use-case here?

Reply:
ok, fair enough. I'll update.